### PR TITLE
ECOPROJECT-5: Documentation for the standalone Node Maintenance Operator

### PR DESCRIPTION
**Pull request by abhatt-rh**
https://issues.redhat.com/browse/TELCODOCS-43 Documentation for the standalone Node Maintenance Operator

Applies to OpenShift version : 4.10 +

Preview:
[Using the Node Maintenance Operator to place nodes in maintenance mode](https://deploy-preview-42107--osdocs.netlify.app/openshift-enterprise/latest/nodes/nodes/eco-node-maintenance-operator.html)

### CODE REVIEW

**Comment by abhatt-rh** [comment]
Hi @razo7 @anna-savina 
Kindly review the documentation for the standalone NMO. Please note that i am working on the installation sections and I will submit them in this PR later today. In the interim, it would help to get reviews on what we have so far.
Thank you!

File: modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+* Install the {product-title} CLI `oc`.
+* Log in as a user with `cluster-admin` privileges.
+
+.Procedure
+
+* Check the status of current node maintenance tasks by running the following command:
++
+[source,terminal]
+----
+$ oc get NodeMaintenance -o yaml
```
Comment (anna-savina) [inline modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc:22]: oc get nm -o yaml

File: modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+----
++
+.Example output
++
+[source,yaml]
+----
+apiVersion: v1
+items:
+- apiVersion: nodemaintenance.kubevirt.io/v1beta1
```
Comment (anna-savina) [inline modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc:31]: apiVersion: nodemaintenance.medik8s.io/v1beta1

File: modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  spec:
+    nodeName: node-1.example.com
+    reason: Node maintenance
+  status:
+    evictionPods: 3   <1>
+    pendingPods:
+    - pod-example-workload-0
+    - httpd
+    - httpd-manual
+    phase: Running
```
Comment (anna-savina) [inline modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc:44]: better to publish example after nodemaintenance succeeded and no pendingPods:
status:
    evictionPods: 3   <1>
    phase: Succeeded

File: modules/eco-resuming-node-from-maintenance-mode-with-cr.adoc [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+[source,terminal]
+----
+$ oc delete -f nodemaintenance-cr.yaml
+----
++
+.Example output
++
+[source,terminal]
+----
+nodemaintenance.nodemaintenance.kubevirt.io "maintenance-example" deleted
```
Comment (anna-savina) [inline modules/eco-resuming-node-from-maintenance-mode-with-cr.adoc:29]: nodemaintenance.nodemaintenance.medik8s.io "maintenance-example" deleted

File: modules/eco-setting-node-to-maintenance-mode-with-cr.adoc [hunk 5]
```diff
@@ -0,0 +1,56 @@
+// Module included in the following assemblies:
+//
+//nodes/nodes/eco-node-maintenance-operator.adoc
+
+:_content-type: PROCEDURE
+[id="eco-setting-node-to-maintenance-mode-with-cr_{context}"]
+= Setting a node to maintenance mode with a NodeMaintenance custom resource
```
Comment (anna-savina) [inline modules/eco-setting-node-to-maintenance-mode-with-cr.adoc:7]: why chapters "Checking status of current NodeMaintenance CR tasks" and "Resume..." were before "Start..."?

File: modules/eco-setting-node-to-maintenance-mode-with-cr.adoc [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+* Install the {product-title} CLI `oc`.
+* Log in to the cluster as a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Create the following node maintenance CR, and save the file as `nodemaintenance-cr.yaml`:
++
+[source,yaml]
+----
+apiVersion: nodemaintenance.kubevirt.io/v1beta1
```
Comment (anna-savina) [inline modules/eco-setting-node-to-maintenance-mode-with-cr.adoc:22]: apiVersion: nodemaintenance.medik8s.io/v1beta1

File: modules/eco-about-node-maintenance-standalone.adoc [hunk 7]
```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+// nodes/nodes/eco-node-maintenance-operator.adoc
+
+:_content-type: CONCEPT
+[id="eco-about-node-maintenance-operator_{context}"]
+= About the Node Maintenance Operator
+
+You can place nodes into maintenance mode using the `oc adm` utility, or using `NodeMaintenance` custom resources (CRs).
+
+Placing a node into maintenance marks the node as unschedulable and drains all the virtual machines and pods from it. Virtual machine instances that have a `LiveMigrate` eviction strategy are live migrated to another node without loss of service. This eviction strategy is configured by default in virtual machine created from common templates but must be configured manually for custom virtual machines.
```
Comment (slintes) [inline modules/eco-about-node-maintenance-standalone.adoc:10]: Virtual machines are not interesting for standalone NMO, because you should not have both at the same time for 4.10. Since we keep the Virtulaization - NMO docs, I would remove all references to VMs here.
For 4.11 we can consider where to place VM specific docs, here or at Virt docs.

Comment (slintes) [reply modules/eco-about-node-maintenance-standalone.adoc:10]: that only leaves `Placing a node into maintenance marks the node as unschedulable and drains all pods from it.` at this place, and then it continues at line 19.

File: modules/eco-about-node-maintenance-standalone.adoc [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+Virtual machine instances without an eviction strategy are shut down. Virtual machines with a `RunStrategy` of `Running` or `RerunOnFailure` are recreated on another node. Virtual machines with a `RunStrategy` of `Manual` are not automatically restarted.
+
+[IMPORTANT]
+====
+Virtual machines must have a persistent volume claim (PVC) with a shared `ReadWriteMany` (RWX) access mode to be live migrated.
+====
+
+When installed as part of {VirtProductName}, Node Maintenance Operator watches for new or deleted `NodeMaintenance` CRs. When a new `NodeMaintenance` CR is detected, no new workloads are scheduled and the node is cordoned off from the rest of the cluster. All pods that can be evicted are evicted from the node. When a `NodeMaintenance` CR is deleted, the node that is referenced in the CR is made available for new workloads.
```
Comment (slintes) [inline modules/eco-about-node-maintenance-standalone.adoc:19]: remove `When installed as part of {VirtProductName},`

File: nodes/nodes/eco-node-maintenance-operator.adoc [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+= Using the Node Maintenance Operator to place nodes in maintenance mode
+include::modules/common-attributes.adoc[]
+:context: node-maintenance-operator
+toc::[]
+
+You can use the Node Maintenance Operator to place nodes under maintenance mode. This is a standalone version of the Node Maintenance Operator that is independent of OpenShift Virtualization installation.
+
+[NOTE]
+====
+If you have installed OpenShift Virtualization, you must use the Node Maintenance Operator that is bundled with it.
```
Comment (slintes) [inline nodes/nodes/eco-node-maintenance-operator.adoc:12]: 👍🏼

File: nodes/nodes/eco-node-maintenance-operator.adoc [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+[NOTE]
+====
+If you have installed OpenShift Virtualization, you must use the Node Maintenance Operator that is bundled with it. 
+====
+
+include::modules/eco-about-node-maintenance-standalone.adoc[leveloffset=+1]
+
+include::modules/eco-maintaining-bare-metal-nodes.adoc[leveloffset=+1]
+
+You can install the Node Maintenance Operator using the web console or the OpenShift CLI (`oc`).
```
Comment (razo7) [inline nodes/nodes/eco-node-maintenance-operator.adoc:19]: No instructions are needed for how to install?

File: nodes/nodes/eco-node-maintenance-operator.adoc [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+
+Place a node into maintenance from the CLI or using a `NodeMaintenance` custom resource.
```
Comment (razo7) [inline nodes/nodes/eco-node-maintenance-operator.adoc:22]: Maybe combine/move this line to the last line (installation).

File: modules/eco-resuming-node-maintenance-cli.adoc [hunk 12]
```diff
@@ -0,0 +1,18 @@
+// Module included in the following assemblies:
+//
+//nodes/nodes/eco-node-maintenance-operator.adoc
+
+:_content-type: PROCEDURE
+[id="eco-resuming-node-maintenance-cli_{context}"]
+= Resuming a node from maintenance mode in the CLI
```
Comment (razo7) [inline modules/eco-resuming-node-maintenance-cli.adoc:7]: If we first show CLI option on how put node unde maintenance, then I would move this subsection right after "Setting a node to maintenance mode in the CLI".

File: nodes/nodes/eco-node-maintenance-operator.adoc (hunk 10 above)
Comment (abhatt-rh) [reply nodes/nodes/eco-node-maintenance-operator.adoc:19]: Hi Or, as I have noted in the comment, the installation instructions are in progress

> Hi @razo7 @anna-savina Kindly review the documentation for the standalone NMO. Please note that I am working on the installation sections and I will submit them in this PR later today. In the interim, it would help to get reviews on what we have so far. Thank you!
> 
> cc: @beekhof

File: nodes/nodes/eco-node-maintenance-operator.adoc (hunk 11 above)
Comment (abhatt-rh) [reply nodes/nodes/eco-node-maintenance-operator.adoc:26]: @razo7 Both are different procedures so can't combine them. With the addition of installation sections, it will work fine

File: nodes/nodes/eco-node-maintenance-operator.adoc (hunk 10 above)
Comment (abhatt-rh) [reply nodes/nodes/eco-node-maintenance-operator.adoc:20]: @razo7 I have now added the installation from web and cli sections

File: modules/eco-setting-node-to-maintenance-mode-with-cr.adoc (hunk 5 above)
Comment (abhatt-rh) [reply modules/eco-setting-node-to-maintenance-mode-with-cr.adoc:7]: @anna-savina I am not sure if i understand the question. Can you please elaborate?

File: modules/eco-setting-node-maintenance-cli.adoc [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+----
+
+. Drain the node in preparation for maintenance.
++
+[source,terminal]
+----
+$ oc adm drain <node1> --delete-local-data --ignore-daemonsets=true --force
+----
+
+* The `--delete-local-data` flag removes any virtual machine instances on the node that use `emptyDir` volumes. Data in these volumes is ephemeral and is safe to be deleted after termination.
```
Comment (abhatt-rh) [inline modules/eco-setting-node-maintenance-cli.adoc:27]: @razo7 @slintes `--delete-local-data` what does this flag do for the standalone NMO? I was trying to remove references to _virtual machine_

File: modules/eco-setting-node-maintenance-cli.adoc [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+
+* The `--ignore-daemonsets=true` flag ensures that daemon sets are ignored and pod eviction can continue successfully.
```
Comment (kquinn1204) [inline modules/eco-setting-node-maintenance-cli.adoc:29]: is it DaemonSets ? Do we have a standard way to define these ? 

DaemonSet and ReplicaSet

File: nodes/nodes/eco-node-maintenance-operator.adoc [hunk 15]
```diff
@@ -0,0 +1,40 @@
+:_content-type: ASSEMBLY
+[id="node-maintenance-operator"]
+= Using the Node Maintenance Operator to place nodes in maintenance mode
+include::modules/common-attributes.adoc[]
+include::modules/virt-document-attributes.adoc[]
+:context: node-maintenance-operator
+toc::[]
+
+You can use the Node Maintenance Operator to place nodes under maintenance mode. This is a standalone version of the Node Maintenance Operator that is independent of OpenShift Virtualization installation.
```
Comment (kquinn1204) [inline nodes/nodes/eco-node-maintenance-operator.adoc:9]: You can use the Node Maintenance Operator to place nodes under maintenance mode

maybe rephrase to 

You can use the Node Maintenance Operator to place nodes in maintenance mode.

or 

You can use the Node Maintenance Operator to place nodes under maintenance.

**Review by kquinn1204 — COMMENTED** [review]
Some minor comments. Fantastic write up with good examples with output which should be easy for customer to follow.

File: modules/eco-maintaining-bare-metal-nodes.adoc [hunk 16]
```diff
@@ -0,0 +1,11 @@
+// Module included in the following assemblies:
+//
+// nodes/nodes/eco-node-maintenance-operator.adoc
+
+[id="eco-maintaining-bare-metal-nodes_{context}"]
+= Maintaining bare metal nodes
+
+When you deploy {product-title} on bare metal infrastructure, there are additional considerations that must be taken into account compared to deploying on cloud infrastructure. Unlike in cloud environments where the cluster nodes are considered ephemeral, re-provisioning a bare metal node requires significantly more time and effort for maintenance tasks.
+
+When a bare metal node fails, for example, if a fatal kernel error happens or a NIC card hardware failure occurs, workloads on the failed node need to be restarted elsewhere else on the cluster while the problem node is repaired or replaced. Node maintenance mode allows cluster administrators to gracefully power down nodes, moving workloads to other parts of the cluster and ensuring workloads do not get interrupted. Detailed progress and node status details are provided during maintenance.
```
[the remark this answers is not in the document]
Comment (abhatt-rh) [reply modules/eco-maintaining-bare-metal-nodes.adoc:10]: revised the para a bit

File: modules/eco-setting-node-maintenance-cli.adoc (hunk 14 above)
Comment (abhatt-rh) [reply modules/eco-setting-node-maintenance-cli.adoc:29]: Yes, seems to align with the guideline for [API object formatting](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting)

File: modules/eco-node-maintenance-operator-installation-cli.adoc [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+You can install the Node Maintenance Operator in your own namespace or in the `openshift-operators` namespace. 
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Log in as a user with `cluster-admin` privileges.
+
+To install the Operator in your own namespace, follow the steps in the procedure. 
+
+To install the Operator in the `openshift-operators` namespace, skip to step 3 of the procedure because the steps to create to create a new `Namespace` custom resource (CR) and a `OperatorGroup` CR are not required.
```
[the remark this answers is not in the document]
Comment (abhatt-rh) [reply modules/eco-node-maintenance-operator-installation-cli.adoc:19]: Yeah, I had used "as" but later when I check the ISG Word Usage for [as](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#as), I changed it to "because"

**Comment by abhatt-rh** [comment]
Hi @razo7 @slintes and @anna-savina
I have incorporated you comments and updated the docs. PTAL again.
Thanks!

File: modules/eco-node-maintenance-operator-installation-cli.adoc [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+.. To create the `Subscription` CR, run the following command:
++
+[source,terminal]
+----
+$ oc create -f node-maintenance-subscription.yaml
+----
+
+.Verification
+
+. Verify that the installation succeeded by inspecting the CSV resource:
```
Comment (jeana-redhat) [inline modules/eco-node-maintenance-operator-installation-cli.adoc:93]: Will CSV be unambiguous to the audience?

File: modules/eco-node-maintenance-operator-installation-cli.adoc [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+----
++
+.Example output
+
+[source,terminal]
+----
+NAME                               DISPLAY                     VERSION   REPLACES   PHASE
+node-maintenance-operator.v0.1.1   Node Maintenance Operator   0.1.1                Succeeded
+----
+. Verify that the Node Maintenance Operator is up and running:
```
Comment (jeana-redhat) [inline modules/eco-node-maintenance-operator-installation-cli.adoc:107]:
```suggestion
. Verify that the Node Maintenance Operator is running:
```

File: nodes/nodes/eco-node-maintenance-operator.adoc [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+
+include::modules/eco-maintaining-bare-metal-nodes.adoc[leveloffset=+1]
+
+You can install the Node Maintenance Operator using the web console or the OpenShift CLI (`oc`).
+
+include::modules/eco-node-maintenance-operator-installation-web-console.adoc[leveloffset=+1]
+
+include::modules/eco-node-maintenance-operator-installation-cli.adoc[leveloffset=+1]
+
+Place a node into maintenance from the CLI.
```
Comment (jeana-redhat) [inline nodes/nodes/eco-node-maintenance-operator.adoc:26]: This seems a bit out of place in the rendered output - it almost looks like a next step for the preceding procedure, but is not formatted as such, and is a bit easy to lose. Consider maybe expanding  it into more of an intro ("You can ....."), or maybe removing it entirely?

Same for lines 30, 34, and 40.

File: modules/eco-setting-node-maintenance-cli.adoc [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+----
+
+. Drain the node in preparation for maintenance.
++
+[source,terminal]
+----
+$ oc adm drain <node1> --delete-local-data --ignore-daemonsets=true --force
+----
+
+* The `--delete-local-data` flag removes any machine instances on the node that use `emptyDir` volumes. Data in these volumes is ephemeral and is safe to be deleted after termination.
```
Comment (jeana-redhat) [inline modules/eco-setting-node-maintenance-cli.adoc:27]:
```suggestion
* The `--delete-local-data` flag removes any machine instances on the node that use `emptyDir` volumes. Data in these volumes is ephemeral and is safe to delete after termination.
```

File: modules/eco-setting-node-to-maintenance-mode-with-cr.adoc [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+kind: NodeMaintenance
+metadata:
+  name: maintenance-example  <1>
+spec:
+  nodeName: node-1.example.com <2>
+  reason: "Node maintenance" <3>
+----
+<1> Node maintenance CR name
+<2> The name of the node to be put into maintenance mode
+<3> Plain text description of the reason for maintenance
```
Comment (jeana-redhat) [inline modules/eco-setting-node-to-maintenance-mode-with-cr.adoc:32]:
```suggestion
<1> The name of the node maintenance CR.
<2> The name of the node to be put into maintenance mode.
<3> A plain text description of the reason for maintenance.
```

File: modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+    reason: Node maintenance
+  status:
+    evictionPods: 3   <1>
+    phase: Succeeded
+    lastError: "Last failure message" <2>
+    totalpods: 5
+...
+----
+<1> `evictionPods` is the number of pods scheduled for eviction.
+<2> `lastError` records the latest eviction error, if any.
```
Comment (jeana-redhat) [inline modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc:46]: Just for consistency, since the parameter names are not included in the annotations above.
```suggestion
<1> The number of pods scheduled for eviction.
<2> The latest eviction error, if any.
```

**Review by jeana-redhat — COMMENTED** [review]
Looks good, just a few very minor suggestions (and some I'm not even sure about, up to you). 

One more impactful thing to think about - right now the structure is:

1. About
2. Maintaining
3. Installing (console)
4. Installing (CLI)
5. Setting (CLI)
6. Resuming (CLI)
7. Setting (console)
8. Resuming (console)

I think it might be worth rearranging into either (something like):
1. About
2. Maintaining
3. CLI: 
  a. Install
  b. Set
  c. Resume
4. Console: 
  a. Install
  b. Set
  c. Resume

=OR=

1. About
2. Maintaining
3. Install
  a. console
  b. CLI
4. Set
  a. console
  b. CLI
5. Resume
  a. console
  b. CLI

File: modules/eco-node-maintenance-operator-installation-cli.adoc [hunk 24]
```diff
@@ -0,0 +1,120 @@
...
+----
+$ oc get csv -n openshift-operators
+----
++
+.Example output
+
+[source,terminal]
+----
+NAME                               DISPLAY                     VERSION   REPLACES   PHASE
+node-maintenance-operator.v0.1.1   Node Maintenance Operator   0.1.1                Succeeded
```
Comment (anna-savina) [inline modules/eco-node-maintenance-operator-installation-cli.adoc:105]: NAME                                                            DISPLAY                        VERSION   REPLACES    PHASE
node-maintenance-operator.v4.10.0   Node Maintenance Operator     4.10.0                            Succeeded

File: modules/eco-setting-node-to-maintenance-mode-with-cr.adoc (hunk 5 above)
Comment (anna-savina) [reply modules/eco-setting-node-to-maintenance-mode-with-cr.adoc:7]: I was talking about structure of document, but I saw better [comment](https://github.com/openshift/openshift-docs/pull/42107#pullrequestreview-894042263) from jeana-redhat. So just ignore this my comment, please

File: modules/eco-about-node-maintenance-standalone.adoc [hunk 25]
```diff
@@ -0,0 +1,15 @@
+// Module included in the following assemblies:
+// nodes/nodes/eco-node-maintenance-operator.adoc
+
+:_content-type: CONCEPT
+[id="eco-about-node-maintenance-operator_{context}"]
+= About the Node Maintenance Operator
+
+You can place nodes into maintenance mode using the `oc adm` utility, or using `NodeMaintenance` custom resources (CRs).
+
+The Node Maintenance Operator watches for new or deleted `NodeMaintenance` CRs. When a new `NodeMaintenance` CR is detected, no new workloads are scheduled and the node is cordoned off from the rest of the cluster. All pods that can be evicted are evicted from the node. When a `NodeMaintenance` CR is deleted, the node that is referenced in the CR is made available for new workloads.
```
Comment (razo7) [inline modules/eco-about-node-maintenance-standalone.adoc:10]: "is detected, no new..." -> "is detected, then no new"

File: modules/eco-node-maintenance-operator-installation-web-console.adoc [hunk 26]
```diff
@@ -0,0 +1,32 @@
...
+
+To confirm that the installation is successful:
+
+. Navigate to the *Operators* -> *Installed Operators* page.
+. Check that the Operator is installed in the `openshift-operators` namespace and that its status is `Succeeded`.
+
+If the Operator is not installed successfully:
+
+. Navigate to the *Operators* -> *Installed Operators* page and inspect the `Status` column for any errors or failures.
+. Navigate to the *Workloads* -> *Pods* page and check the logs in any pods in the `openshift-operators` project that are reporting issues.
```
Comment (razo7) [inline modules/eco-node-maintenance-operator-installation-web-console.adoc:32]: `openshift-operators` is a namespace, and we have been referring him as namespace in the other occurrences.

File: modules/eco-node-maintenance-operator-installation-cli.adoc [hunk 27]
```diff
@@ -0,0 +1,120 @@
...
+:_content-type: PROCEDURE
+[id="installing-maintenance-operator-using-cli_{context}"]
+= Installing the Node Maintenance Operator by using the CLI
+You can use the OpenShift CLI (`oc`) to install the Node Maintenance Operator.
+
+You can install the Node Maintenance Operator in your own namespace or in the `openshift-operators` namespace. 
+
+To install the Operator in your own namespace, follow the steps in the procedure. 
+
+To install the Operator in the `openshift-operators` namespace, skip to step 3 of the procedure because the steps to create a new `Namespace` custom resource (CR) and an `OperatorGroup` CR are not required.
```
Comment (razo7) [inline modules/eco-node-maintenance-operator-installation-cli.adoc:14]: To install the Operator in the `openshift-operators` namespace, skip to step 3 of the procedure because steps 1 and 2, creating new `Namespace` custom resource (CR), and an `OperatorGroup` CR, are not required.

File: modules/eco-node-maintenance-operator-installation-web-console.adoc (hunk 26 above)
Comment (razo7) [reply modules/eco-node-maintenance-operator-installation-web-console.adoc:32]: Or maybe it is ok, since you might have referred to the project on the top left (where you select the namespace).

File: modules/eco-setting-node-maintenance-cli.adoc [hunk 28]
```diff
@@ -0,0 +1,31 @@
+// Module included in the following assemblies:
+//
+//nodes/nodes/eco-node-maintenance-operator.adoc
+
+:_content-type: PROCEDURE
+[id="eco-setting-node-maintenance-cli_{context}"]
+= Setting a node to maintenance mode in the CLI
```
Comment (razo7) [inline modules/eco-setting-node-maintenance-cli.adoc:7]: "Setting a node to maintenance mode" -> "Setting a node into maintenance mode" ?

Same as for "Setting a node to maintenance mode with a NodeMaintenance custom resource" occurrence

File: modules/eco-setting-node-to-maintenance-mode-with-cr.adoc [hunk 29]
```diff
@@ -0,0 +1,56 @@
...
++
+[source,yaml]
+----
+apiVersion: nodemaintenance.medik8s.io/v1beta1
+kind: NodeMaintenance
+metadata:
+  name: maintenance-example  <1>
+spec:
+  nodeName: node-1.example.com <2>
+  reason: "Node maintenance" <3>
```
Comment (razo7) [inline modules/eco-setting-node-to-maintenance-mode-with-cr.adoc:28]: reason: "NIC replacement".

A more concrete reason I guess.

File: modules/eco-setting-node-to-maintenance-mode-with-cr.adoc [hunk 30]
```diff
@@ -0,0 +1,56 @@
...
+<3> A plain text description of the reason for maintenance
++
+.  Apply the node maintenance schedule by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f nodemaintenance-cr.yaml
+----
+
+. Check the progress of the maintenance task by running the following command, replacing `<node-name>` with the name of your node:
```
Comment (razo7) [inline modules/eco-setting-node-to-maintenance-mode-with-cr.adoc:41]: `<node-name>` -> nodeName

Since nodeName is indeed the node which we put under maintenance

File: modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc [hunk 31]
```diff
@@ -0,0 +1,46 @@
...
+items:
+- apiVersion: nodemaintenance.medik8s.io/v1beta1
+  kind: NodeMaintenance
+  metadata:
+...
+  spec:
+    nodeName: node-1.example.com
+    reason: Node maintenance
+  status:
+    evictionPods: 3   <1>
```
Comment (razo7) [inline modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc:39]: The order of the fields is incorrect IIRC and maybe displaying the status while it is running is more insightful.

evictionPods: 3
lastError: 'Last failure message'
 pendingPods:
-- pod-A
-- pod-B
-- pod-C
 phase: Running
  totalpods: 5

File: modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc [hunk 32]
```diff
@@ -0,0 +1,46 @@
...
+    phase: Succeeded
+    lastError: "Last failure message" <2>
+    totalpods: 5
```
Comment (razo7) [inline modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc:42]: totalpods: 5 <3>

File: modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc [hunk 33]
```diff
@@ -0,0 +1,46 @@
...
+...
+----
+<1> The number of pods scheduled for eviction.
+<2> The latest eviction error, if any.
```
Comment (razo7) [inline modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc:46]: <3> The number of pods before it entered maintenance mode.

File: modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc (hunk 31 above)
Comment (abhatt-rh) [reply modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc:39]: Per our discussion, retaining the `Succeeded` phase

File: modules/eco-node-maintenance-operator-installation-cli.adoc (hunk 27 above)
Comment (abhatt-rh) [reply modules/eco-node-maintenance-operator-installation-cli.adoc:14]: As discussed, retaining the current format

File: modules/eco-maintaining-bare-metal-nodes.adoc [hunk 34]
```diff
@@ -0,0 +1,11 @@
+// Module included in the following assemblies:
+//
+// nodes/nodes/eco-node-maintenance-operator.adoc
+
+[id="eco-maintaining-bare-metal-nodes_{context}"]
+= Maintaining bare-metal nodes
+
+When you deploy {product-title} on bare-metal infrastructure, you must take additional considerations into account compared to deploying on cloud infrastructure. Unlike in cloud environments where the cluster nodes are considered ephemeral, re-provisioning a bare-metal node requires significantly more time and effort for maintenance tasks.
+
+When a bare-metal node fails due to a kernel error or a NIC card hardware failure, workloads on the failed node need to be restarted on another node in the cluster while the problem node is repaired or replaced. Node maintenance mode allows cluster administrators to gracefully turn-off nodes, moving workloads to other parts of the cluster, and ensuring workloads do not get interrupted. Detailed progress and node status details are provided during maintenance.
```
[the remark this answers is not in the document]
Comment (abhatt-rh) [reply modules/eco-maintaining-bare-metal-nodes.adoc]: I have revised the sentence for clarity.

File: modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc [hunk 35]
```diff
@@ -0,0 +1,47 @@
...
+You can check the status of current `NodeMaintenance` CR tasks.
+
+.Prerequisites
+
+* Install the {product-title} CLI `oc`.
+* Log in as a user with `cluster-admin` privileges.
+
+.Procedure
+
+* Check the status of current node maintenance tasks by running the following command:
```
Comment (razo7) [inline modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc:18]: (I would add something about what is `nm`).
Check the current status of `NodeMaintenance` (or it's acronym `nm`) CR  tasks by running the following command:

File: modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc [hunk 36]
```diff
@@ -0,0 +1,47 @@
...
+  status:
+    evictionPods: 3   <1>
+    lastError: "Last failure message" <2>
+    phase: Succeeded
+    totalpods: 5 <3>
+...
+----
+<1> The number of pods scheduled for eviction.
+<2> The latest eviction error, if any.
+<3> The total number of pods before the node entered smaintenance mode.
```
Comment (razo7) [inline modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc:47]: The number of pods before the node entered maintenance mode.

File: modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc (hunk 35 above)
Comment (razo7) [reply modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc:18]: My only concern with my comment is the use of the world "acronym" where it mostly related to capital letters abbreviation and not lower case...

File: _topic_maps/_topic_map.yml [hunk 37]
```diff
@@ -1863,6 +1863,8 @@ Topics:
     File: eco-poison-pill-operator
   - Name: Deploying node health checks by using the Node Health Check Operator
     File: eco-node-health-check-operator
+  - Name: Using the Node Maintenance Operator to place nodes in maintenance mode
```
[the remark this answers is not in the document]
Comment (abhatt-rh) [reply _topic_maps/_topic_map.yml:1885]: Need to retain it as discussed

File: modules/eco-setting-node-maintenance-cli.adoc (hunk 13 above)
Comment (slintes) [reply modules/eco-setting-node-maintenance-cli.adoc]: That flag is deprecated in the meanwhile, it's named `--delete-emptydir-data` now:
> --delete-emptydir-data=false: Continue even if there are pods using emptyDir (local data that will be deleted when
the node is drained)

I'd suggest:

```
$ oc adm drain <node1> --delete-emptydir-data --ignore-daemonsets=true --force
----

* The `--delete-local-data` flag removes any pods on the node that use `emptyDir` volumes. Data in these volumes is ephemeral and is safe to be deleted after termination.
```

File: modules/eco-setting-node-to-maintenance-mode-with-cr.adoc [hunk 38]
```diff
@@ -0,0 +1,56 @@
+// Module included in the following assemblies:
+//
+//nodes/nodes/eco-node-maintenance-operator.adoc
+
+:_content-type: PROCEDURE
+[id="eco-setting-node-to-maintenance-mode-with-cr_{context}"]
+= Setting a node to maintenance mode with a NodeMaintenance CR
+
+You can put a node into maintenance mode with a `NodeMaintenance` custom resource (CR). When you apply a `NodeMaintenance` CR, all allowed pods are evicted and the node is shut down. Evicted pods are queued to be moved to another node in the cluster.
```
Comment (anna-savina) [inline modules/eco-setting-node-to-maintenance-mode-with-cr.adoc:9]: node isn't shut down, node is unschedulable

File: modules/eco-setting-node-to-maintenance-mode-with-cr.adoc [hunk 39]
```diff
@@ -0,0 +1,56 @@
...
+  name: maintenance-example  <1>
+spec:
+  nodeName: node-1.example.com <2>
+  reason: "NIC replacement" <3>
+----
+<1> The name of the node maintenance CR
+<2> The name of the node to be put into maintenance mode
+<3> A plain text description of the reason for maintenance
++
+.  Apply the node maintenance schedule by running the following command:
```
Comment (anna-savina) [inline modules/eco-setting-node-to-maintenance-mode-with-cr.adoc:34]: Optionally. Maybe better to say:
Apply the node maintenance CR by running the following command

File: modules/eco-resuming-node-from-maintenance-mode-with-cr.adoc [hunk 40]
```diff
@@ -0,0 +1,30 @@
...
+
+.Procedure
+
+* When your node maintenance task is complete, delete the active `NodeMaintenance` CR:
++
+[source,terminal]
+----
+$ oc delete -f nodemaintenance-cr.yaml
+----
++
```
Comment (anna-savina) [inline modules/eco-resuming-node-from-maintenance-mode-with-cr.adoc:24]: Optionally:

or
oc delete nm <maintenance-example>

Comment (razo7) [reply modules/eco-resuming-node-from-maintenance-mode-with-cr.adoc:24]: I think it is better as it was.

**Comment by abhatt-rh** [comment]
@jeana-redhat @razo7 @anna-savina 
I have incorporated all applicable comments. If everything looks ok, could you provide your acks/lgtm?
